tests/api2/test_zfs_snapshot_hold.py

```diff
@@ -0,0 +1,20 @@
+from unittest.mock import ANY
+
+from middlewared.test.integration.assets.pool import dataset, snapshot
+from middlewared.test.integration.utils import call
+
+
+def test_normal_snapshot():
+    with dataset("test") as ds:
+        with snapshot(ds, "test") as id:
+            assert call("zfs.snapshot.get_instance", id, {"extra": {"holds": True}})["holds"] == {}
+
+
+def test_held_snapshot():
+    with dataset("test") as ds:
```

Comment:
Maybe more descriptive dataset name for test. Less likely for collision if someone else uses `test` and doesn't properly clean up.

Reply:
Do we need test for recursive ops as well?

Reply:
> if someone else uses test and doesn't properly clean up.

then they will be exposed and they will have to clean up properly